### PR TITLE
Get kubectl config file from k8s vagrant dev vm

### PR DESCRIPTION
**Pull request by nebril**
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/5066)

### CODE REVIEW

**Comment by nebril** [comment]
test-docs-please

**Comment by raybejjani** [comment]
Would it be more flexible to do what minikube does and add this to the `$HOME/.kube/config` file? (or tell the user to add it with the ssh | cat | sed thing). This would allow a developer to use the kubectl commands without too much weirdness (I just discovered a whole world of cluster/context management commands there)

File: Documentation/contributing.rst [hunk 1]
```diff
@@ -136,6 +136,8 @@ kubernetes installed and plus a worker, run:
 
 	$ RUNTIME=containerd K8S=1 NWORKERS=1 contrib/vagrant/start.sh
 
+If you want to connect to dev k8s cluster via kubectl from your host machine, set ``KUBECONFIG`` env var to ``$GOPATH/src/github.com/cilium/cilium/vagrant.kubeconfig`` and add ``127.0.0.1 k8s1`` line to your hosts file.
```
Comment (ianvernon) [inline Documentation/contributing.rst]: Avoid using abbreviations in documentation.

"kubectl" --> "\`\`kubectl\`\`\"
"connect to dev k8s cluster" --> "connect to the developer VM provisioned with Kubernetes"
"env var" --> "environment variable"
"line to your hosts file" --> "to your hosts file"

File: Documentation/spelling_wordlist.txt [hunk 2]
```diff
@@ -128,6 +128,7 @@ endian
 endianness
 endpointSelector
 enum
+env
```
Comment (ianvernon) [inline Documentation/spelling_wordlist.txt]: Remove this.

**Review by ianvernon — CHANGES_REQUESTED** [review]
Main change looks good, but have some documentation nitpicks.

**Comment by nebril** [comment]
@raybejjani it's certainly possible, I thought about this, but decided against it to spend less time on this PR. We can file an issue for that.

File: Documentation/contributing.rst (hunk 1 above)
Comment (nebril) [reply Documentation/contributing.rst]: Thanks. Changed the "connect" section point out that he is connecting to k8s cluster.

File: Documentation/spelling_wordlist.txt (hunk 2 above)
Comment (nebril) [reply Documentation/spelling_wordlist.txt]: done

**Comment by nebril** [comment]
test-docs-please

**Comment by nebril** [comment]
@ianvernon please re-review

**Comment by raybejjani** [comment]
@nebril Isn't it just saying add this snippet to your kubectl config file instead of making the new file and editing the hosts file? The cluster name in that config is distinct from the server and we already seem to know the IP of the server (loopback since we're forwarding ports)

**Comment by raybejjani** [comment]
@nebril Isn't it just saying add this snippet to your kubectl config file instead of making the new file and editing the hosts file? The cluster name in that config is distinct from the server and we already seem to know the IP of the server (loopback since we're forwarding ports).
I'm worried that we'll introduce this non-canonical way of wiring kubectl and have to maintain it forever, never knowing if anyone uses it. The effort to do it more properly seems worth it but I might be wrong that adding things to the kubectl config is the correct way.

**Comment by nebril** [comment]
@raybejjani there are a lot of ways to configure multi cluster access, setting KUBECONFIG seems like the easiest one to do without interfering with user setup too much. If you have other per-cluster config files you should append `vagrant.kubeconfig` to your `KUBECONFIG`, but at this point you probably know what you're doing: https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/#set-the-kubeconfig-environment-variable

I can add info about appending this filepath to your kubeconfig, I still think this will be the easiest path.

**Comment by nebril** [comment]
test-docs-please

**Review by ianvernon — APPROVED** [review]

**Pull request merged**